src/v/redpanda/application.cc

```diff
@@ -2013,13 +2013,7 @@ void application::wire_up_redpanda_services(
         vassert(
           archival_storage_enabled(),
           "cloud topics currently requires archival storage to be enabled");
-        construct_service(
-          _reconciler,
-          &partition_manager,
-          &cloud_io,
-          cloud_storage_clients::bucket_name(
-            config::shard_local_cfg().cloud_storage_bucket().value()))
-          .get();
```

Comment:
FWIW once https://github.com/redpanda-data/redpanda/pull/23758/commits/8d82e2ad34f9674c02b7af02a9f80f459f5bc916 is in, it may be worth moving construction around so we all use the same cloud config

Reply:
Oops i meant this commit https://github.com/redpanda-data/redpanda/pull/23758/commits/92a155c4a0c99a2e32ce43510b9a94485c651975

Reply:
Agh, neither link shows the PR, which is here https://github.com/redpanda-data/redpanda/pull/23758

Reply:
oh cool thanks. yes, i will refactor then